### DISCUSSION TITLE
How to merge Two graphs' edge features?

### DISCUSSION THREAD

**Discussion by StefanIsSmart** [pyg-team/pytorch_geometric#8800] [Q&A]
Sometimes, I have two graphs（graph_1 and graph_2）. They have the same nodes, but their edge_index and edge_features are different, and I want to merge them to get their union. 
That means the edge_index will merge as the union, and the edge features will merge too (torch.concat).
If graph_1 has the features but graph_2 doesn't have, it will be padding as 0.

Does pyg have a util like that?

**Comment by rusty1s** [comment]
Take a look at `utils.coalesce`:
```
edge_index = torch.cat([edge_index1, edge_index1], dim=1)
edge_feat = torch.cat([edge_feat1, edge_feat2], dim=0)

edge_index, edge_feat = coalesce(edge_index, edge_feat)
